search/engine.go

```diff
@@ -607,6 +607,10 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 	defer e.timeTrack(time.Now(), consts.LAT_DOSEARCH)
 
 	suggestChannel := make(chan null.String)
+	grammarsSingleHitIntentsChannel := make(chan []Intent, 1)
+	grammarsFilterIntentsChannel := make(chan []Intent, 1)
+	grammarsFilteredResultsByLangChannel := make(chan map[string]FilteredSearchResult)
```

Comment:
Missing channel size to be 1, othewise the "recover" phases will stack on this channel write as no one reads from this channel yet.

Reply:
The recover is successful after panic even without setting channel size to  grammarsFilteredResultsByLangChannel